### PR TITLE
Support Conan

### LINKED ISSUE #24: Support Conan package
What?  
Include Conan.io support to this project

Why?  
Conan is an opportunity to distribute this project as a package.

How?  
- Include conanfile.py to package this project
- Include test_package to validate the recipe
- Include build.py to run CPT
- Include Job on Travis CI
- Include job on Appveyor

### PR DESCRIPTION
**Pull request by uilianries**
Hi!

This PR brings the support to build taocpp-json as Conan package

What I included:

- Conan recipe as did for PEGTL and operators
- Run Conan package on Travis CI (linux) and Appveyor
- Add test package to validate the Conan package
- Add build.py to help Conan during CI stage

What should be done before to merge this branch:

- Set CONAN_LOGIN_USERNAME on Travis settings with `taocpp-user`
- Set CONAN_PASSWORD on Travis settings with taocpp-user API Key

/cc @pleroux0

Close #24

### CODE REVIEW

**Comment by uilianries** [comment]
@d-frey and @pleroux0

What do you think about rename the target to `taocpp::json` ?

**Comment by d-frey** [comment]
I took care of CONAN_LOGIN_USERNAME and CONAN_PASSWORD.

Yes, please rename to taocpp::json.

I also committed to taocpp/operators some stylistic fixes, repeating expressions in CMake's else()/endif() is superfluous and endforeach() only needs the variable name, not the whole expression.

Finally, taocpp/json does not yet have official releases, but we are releasing beta versions right now. Would it be possible to also have an automatic upload for these versions to be recognized and uploaded as `json/<VERSION>@taocpp/testing`? (the pattern for these versions is `\d+\.\d+\.\d+-.+`)

This would also allow us to test the whole Conan packaging chain outside of /stable.

**Comment by uilianries** [comment]
> I took care of CONAN_LOGIN_USERNAME and CONAN_PASSWORD.
Thanks!

> Yes, please rename to taocpp::json.
Okay, I'll take care

> I also committed to taocpp/operators some stylistic fixes, repeating expressions in CMake's else()/endif() is superfluous and endforeach() only needs the variable name, not the whole expression.
Okay, I'll do the same for json

> Finally, taocpp/json does not yet have official releases, but we are releasing beta versions right now. Would it be possible to also have an automatic upload for these versions to be recognized and uploaded as json/<VERSION>@taocpp/testing? (the pattern for these versions is \d+\.\d+\.\d+-.+)
Yes, it's totally possible. I'm going to apply a rule to just upload when using tag or under master branch. For master it will be uploaded to testing channel, otherwise it will be stable.

**Comment by uilianries** [comment]
BTW I just created the json repo on Bintray: https://bintray.com/taocpp/public-conan/json:taocpp

There is no packages yet, but all package profile is done.

**Comment by d-frey** [comment]
> Yes, it's totally possible. I'm going to apply a rule to just upload when using tag or under master branch. For master it will be uploaded to testing channel, otherwise it will be stable.

I'm not sure I understand that part. The testing-releases will also have a tag, so we only need to upload tags. The tag's pattern should decide whether it goes to stable (`v?\d+\.\d+\.\d+`) or testing (`v?\d+\.\d+\.\d+-.*`).

Note I also fixed the patterns: They are now what they should probably be for all projects following semantic versioning. They have an optional `v` at the start (as some people tag with `v`, some (us) without) and the dot is also escaped so you won't accidentally match `0a0b0`. I feel like this would be general enough (SemVer support) that some of the code could be moved to Conan (or the CPT) so everyone can benefit from it and it would keep the project's files simpler.

File: .conan/build.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ def upload(self):
         ci_manager = CIManager(printer)
         branch = ci_manager.get_branch()
 
-        patterns = ["master$", self.stable_branch_pattern]
+        patterns = ["v?\d+\.\d+\.\d+-.*", self.stable_branch_pattern]
```
Comment (d-frey) [inline .conan/build.py]: I'm not a Python expert, but I think this line is missing an `r` before the string.

Comment (uilianries) [reply .conan/build.py]: Ops! Yeah, you are right.

**Pull request merged**